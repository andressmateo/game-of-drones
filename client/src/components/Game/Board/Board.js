import React from 'react';

const Board = ({ player: { name }, options, play }) => {
  return (
    <div>
      <h3>{name}</h3>
      <br />
      {options.map(option => (
        <button onClick={() => play(option)} key={option}>
          {option}
        </button>
      ))}
    </div>
  );
};

export default Board;

/*
{game:{
    turns:[
      {
        move1:"",
        move2:"",
        winner:"",
        isOver
      }
    ],
    winner:"",
    isOver:""
}};

{ move: "paper", kills: "rock"},
{ move: "rock", kills: "scissors"},
{ move: "scissors", kills: "paper"}
*/
