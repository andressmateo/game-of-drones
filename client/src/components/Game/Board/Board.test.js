import React from 'react';
import { shallow } from 'enzyme';
import chai, { expect } from 'chai';
import renderer from 'react-test-renderer';
import chaiEnzyme from 'chai-enzyme';
import chaiJestSnapshot from 'chai-jest-snapshot';
import sinon from 'sinon';

import Board from './Board';

chai.use(chaiEnzyme());
chai.use(chaiJestSnapshot);

const props = { player: { name: 'player1' }, options: [] };

describe('Board component', () => {
  it('renders without crashing', () => {
    shallow(<Board {...props} />);
  });

  it('renders correctly - snapshot', () => {
    const tree = renderer.create(<Board {...props} />);
    expect(tree).to.matchSnapshot();
  });
});
