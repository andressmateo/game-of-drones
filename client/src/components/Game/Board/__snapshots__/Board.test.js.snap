// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component renders correctly - snapshot 1`] = `
<div>
  <h2>
    player1
  </h2>
  <span>
    Select your move: 
  </span>
  <div
    className="sc-bxivhb cWWhYj"
  />
</div>
`;
